server/cmd/dcrdex/main.go

```diff
@@ -82,15 +82,17 @@ func mainCore(ctx context.Context) error {
 	// Load, or create and save, the DEX signing key.
 	var privKey *secp256k1.PrivateKey
 	{
-		keyPW, err := admin.PasswordPrompt("Signing key password: ")
-		if err != nil {
-			return fmt.Errorf("cannot use password: %v", err)
+		if len(cfg.SigningKeyPW) == 0 {
+			cfg.SigningKeyPW, err = admin.PasswordPrompt(ctx, "Signing key password: ")
+			if err != nil {
+				return fmt.Errorf("cannot use password: %v", err)
+			}
 		}
-		privKey, err = dexKey(cfg.DEXPrivKeyPath, keyPW)
+		privKey, err = dexKey(cfg.DEXPrivKeyPath, cfg.SigningKeyPW)
+		admin.ClearBytes(cfg.SigningKeyPW)
 		if err != nil {
 			return err
 		}
-		admin.ClearBytes(keyPW)
 	}
```

Comment:
This was already here, but I'm confused as to why this is a block. Is there a golang thing I'm not aware of?

Reply:
It's not, I just moved the clearing earlier, to line 92, immediately after `keyPW` is used in line 91.

Reply:
@JoeGruffins is referring to the extra scope `{ ... }` after `var privKey *secp256k1.PrivateKey`.  That can be removed now.  I had that there to limit scope of `keyPW`, whihc is gone.

Reply:
Oh. Got it.